From 60629f48f5e143de8fe52592cd428bcd0abcc36b Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Tue, 4 Dec 2012 16:21:02 +0200
Subject: [PATCH 435/518] dpa_classifier: Add type cast to avoid build warning

In function "flush_table" the internal variable "i" is an unsigned
integer. When used in the "FM_PCD_MatchTableModifyNextEngine" function
call it is automatically cast to a 16bits unsigned integer. To make it
clear that this aspect was analyzed and there is no danger of data loss,
a specific type cast was added.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index 136a0193..da2aad0 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -1731,7 +1731,7 @@ static int flush_table(struct dpa_cls_table *ptable)
 		for (i = 0; i < ptable->params.indexed_params.entries_cnt;
 									i++) {
 			err = FM_PCD_MatchTableModifyNextEngine(cc_node,
-							i,
+							(uint16_t)i,
 							&next_engine_params);
 			if (err != E_OK) {
 				pr_err("ERROR: %s, %s (%d): FMan driver call "
-- 
1.7.5.4

